lib/WebService/Async/CustomerIO.pm

```diff
@@ -0,0 +1,266 @@
+package WebService::Async::CustomerIO;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::CustomerIO - unofficial support for the Customer.io service
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+=cut
+
+use parent qw(IO::Async::Notifier);
+
+use Syntax::Keyword::Try;
+use Future;
+use Net::Async::HTTP;
+use Carp qw();
+use JSON::MaybeUTF8 qw(:v1);
+
+use WebService::Async::CustomerIO::Customer;
+use WebService::Async::CustomerIO::RateLimiter;
+use WebService::Async::CustomerIO::Trigger;
+
+use constant {
+    TRACKING_END_POINT => 'https://track.customer.io/api/v1',
+    API_END_POINT      => 'https://api.customer.io/v1/api',
+    RPS_LIMIT_TRACKING => 30,
+    RPS_LIMIT_API      => 10,
+};
+
+=head2 new(%params) -> obj
+
+Creates a new api client object
+
+Parameters:
+- site_id
+- api_key
+
+=cut
+
+sub configure {
+    my ($self, %args) = @_;
+    for my $k (qw(site_id api_key)) {
+        Carp::croak "Missing requeread argument: $k" unless exists $args{$k};
+        $self->{$k} = delete $args{$k} if exists $args{$k};
+    }
+
+    $self->{tracking_ratelimiter} =
+        WebService::Async::CustomerIO::RateLimiter->new(
+            limit    => RPS_LIMIT_TRACKING,
+            interval => 1,
+        );
+    $self->add_child($self->{tracking_ratelimiter});
+
+    $self->{api_ratelimiter} =
+        WebService::Async::CustomerIO::RateLimiter->new(
+            limit    => RPS_LIMIT_API,
+            interval => 1,
+        );
+    $self->add_child($self->{api_ratelimiter});
+
+    $self->SUPER::configure(%args);
+}
+
+=head2 site_id
+
+=cut
+
+sub site_id {shift->{site_id}}
+
+=head2 api_key
+
+=cut
+
+sub api_key {shift->{api_key}}
+
+=head2 api_ratelimiter
+
+Getter returns RateLimmiter for regular API endpoint.
+
+=cut
+
+sub api_ratelimiter {shift->{api_ratelimiter}}
+
+=head2 tracking_ratelimiter
+
+Getter returns RateLimmiter for tracking API endpoint.
+
+=cut
+
+sub tracking_ratelimiter {shift->{tracking_ratelimiter}}
+
+=head2 tracking_request($method, $uri, $data) -> future($data)
```

Comment:
This makes links much harder to write. Better to have just the method name, so that people can use `L</tracking_request>`, and move the parameters/example usage into the body of the POD.

Reply:
fixed.